### PR TITLE
AsciiDoc how-to

### PR DESCRIPTION
**Pull request by polandll**
This how-to is intended for more than one repo, but I started with Stargate Docs to isolate it.

Review at: https://merckx.sjc.dsinternal.org/en/stargate/stargate/1.0/contribute/

### CODE REVIEW

File: modules/contribute/pages/install-git-and-editor.adoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
+= Install Git and an AsciiDoc Editor
+:url-atom-docs: https://flight-manual.atom.io/getting-started/sections/installing-atom
+:url-linux: {url-atom-docs}/#platform-linux
+:url-mac: {url-atom-docs}/#platform-mac
+:url-windows: {url-atom-docs}/#platform-windows
+
+== Create a GitHub account
+
+The Stargate documentation is stored on GitHub, so if you don't have a GitHub account, you need to https://github.com/join[create one^].
```
Comment (jgillenwater) [inline modules/contribute/pages/install-git-and-editor.adoc:9]:
```suggestion
The Stargate documentation is stored on GitHub. If you don't have a GitHub account, you need to https://github.com/join[create one^].
```

File: modules/contribute/nav.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
+.Contributing to the Documentation
+* xref:index.adoc[Overview]
+
+* Setting Up Your Workspace
+ ** xref:install-git-and-editor.adoc[Install Git and an AsciiDoc editor]
+ ** xref:set-up-repository.adoc[Set up Your local repository]
```
Comment (jgillenwater) [inline modules/contribute/nav.adoc:6]:
```suggestion
 ** xref:set-up-repository.adoc[Set up your local repository]
```

File: modules/contribute/pages/set-up-repository.adoc [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+After you fork the repository, clone it to your device.
+
+. Create or select the folder where you want to store the repository on your device.
+. Open a terminal and change into the selected folder.
+
+ $ cd path/to/folder
+
+. Go to your repository on GitHub, and click btn:[Clone or download].
+. Click the Clipboard icon to copy the repository's Git URL.
```
Comment (jgillenwater) [inline modules/contribute/pages/set-up-repository.adoc:36]: Make "Clipboard icon" lowercase throughout the docs.

File: modules/contribute/pages/set-up-repository.adoc [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+. In your terminal type `git remote add upstream`, paste the copied URL, and press kbd:[Enter].
+For example, to connect to the remote upstream stargate/docs repository, you'd enter:
+
+ $ git remote add upstream git@github.com:stargate/docs.git
+
+. View the configured upstream and origin values and confirm the URLs are correct.
++
+--
+ $ git remote -v
```
Comment (jgillenwater) [inline modules/contribute/pages/set-up-repository.adoc:60]: Provide an example result that could be returned.

File: modules/contribute/pages/repositories.adoc [hunk 5]
```diff
@@ -0,0 +1,30 @@
+= Repository Organization
+:url-git-sg:  https://github.com/stargate
+:url-git-sgdocs: {url-git-sg}/docs
+:url-git-sgantora-ui: {url-git-sg}/antora-ui-stargate
+:url-git-sgdocker: {url-git-sg}/docker-images
+
+//Stargate uses Antora to manage and build stargate.io/docs.
+
+The Stargate documentation is currently stored in a single repository that is
```
Comment (jgillenwater) [inline modules/contribute/pages/repositories.adoc:9]:
```suggestion
The Stargate documentation is stored in a single repository that is
```

File: modules/contribute/pages/repositories.adoc [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+organized in a standard directory structure. This may change in the future.
+
+[#repo-urls]
+// LLP 10.08.21 Not currently used, but when we break out the examples, we'll have
+// multiple repos.
+== Documentation Repository Roster
+
+The following repositories accept contributions via the regular workflow and use a standard branch pattern.
+
+* {url-git-sgdocs}[Stargate Docs^]
```
Comment (jgillenwater) [inline modules/contribute/pages/repositories.adoc:19]: Can we add the K8ssandra docs here, too?

File: modules/contribute/partials/component-terms.adoc [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+All of the files in a component are versioned together.
++
+--
+While it is common to find all of a component's files in a single repository, a
+component can be distributed across several repositories.
+For instance, in the future, we may aggregate the Developer Guide component from
+_developers_guide_ and _code_examples_ repositories.
+
+A component isn't recognized by a repository or directory.
+Instead, the presence of a file named _antora.yaml_ indicates that the directories
```
Comment (jgillenwater) [inline modules/contribute/partials/component-terms.adoc:19]: Do you typically italicize file names? I would've thought we would use `antora.yaml`.

File: modules/contribute/partials/component-terms.adoc [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+and files nested under it are part of a component.
+--
+
+Component Descriptor::
+A component must contain a component descriptor file named _antora.yaml_.
+When Antora finds _antora.yaml_ in a repository, it knows it has located a component (or part of component).
```
Comment (jgillenwater) [inline modules/contribute/partials/component-terms.adoc:25]:
```suggestion
When Antora finds _antora.yaml_ in a repository, it knows it has located a component or part of component.
```

File: modules/contribute/pages/create-branches.adoc [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+[#base-branch]
+== Choose the Right Base Branch
+
+A base branch contains the original source documentation.
+You create working branches from a base branch and send pull requests to a base branch for integration.
+A working branch isolates changes that solve a single issue.
+A pull request submitted from a working branch that only contains closely related changes is easier to test, review, and merge.
+
+// LLP 12.13.21 CHANGE TO MAIN
+The Stargate documentation repositories maintain one base branch: master.
```
Comment (jgillenwater) [inline modules/contribute/pages/create-branches.adoc:15]: Add a link for the repo.

File: modules/contribute/pages/create-branches.adoc [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+At the current time, we only maintain a single version of the Stargate docs.
+
+[#work-branch-atom]
+== Create a Working Branch in Atom
+
+These instructions assume you've xref:install-git-and-editor.adoc#install-atom[installed Atom] and xref:set-up-repository.adoc[forked, cloned, and configured the upstream and origin remotes] of a documentation repository.
```
Comment (jgillenwater) [inline modules/contribute/pages/create-branches.adoc:22]: Could we just use a prereq section with a list?

File: modules/contribute/pages/create-branches.adoc [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+
+. Create a new working branch from the base branch.
+Click the image:git-branch.svg[,16,role=icon] branch icon.
+In the modal window, click btn:[New Branch], and enter the name of the new working branch in the form field.
++
+--
+image::new-branch.png[,380,align=left]
+
+The name of a working branch should pair it with the issue it resolves.
+For example, if you're resolving stargate/docs issue #6500, you could name the working branch `sgdocs-6500`.
```
Comment (jgillenwater) [inline modules/contribute/pages/create-branches.adoc:47]:
```suggestion
For example, if you're resolving `stargate/docs` issue #6500, you could name the working branch `sgdocs-6500`.
```

File: modules/contribute/pages/create-branches.adoc [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+ $ git pull upstream name-of-base-branch
+
+. Create a working branch from the base branch and checkout the working branch.
++
+--
+ $ git checkout -b sgdocs-####
+
+The name of a working branch should pair it with the issue it resolves.
+For example, if you're resolving stargate/docs issue #6500, you could name the working branch `sgdocs-6500`.
```
Comment (jgillenwater) [inline modules/contribute/pages/create-branches.adoc:75]:
```suggestion
For example, if you're resolving `stargate/docs` issue #6500, you could name the working branch `sgdocs-6500`.
```

**Review by jgillenwater — COMMENTED** [review]
Reviewed through "Create a working branch" in the doc navigation. 

The capitalization for section headers and titles are not consistent with the capitalization of the rest of the Stargate or other DS docs. Please review.

**Comment by annieden** [comment]
I have couple of general suggestions.  It is a good policy, especially from a translation standpoint, to avoid using contractions.  Examples are [don't, you're, Let's]... and those are just from the 'Edit existing documentation' page.  The other suggestion is to use the follow-on 'then' after a sentence of clause that begins with 'If'.  For instance, under the heading 'General Editing Steps Using Atom', bullet 2a: "If you’re not familiar with the structure of a documentation component, see Directory Structure and Key Files."   To avoid the If, then clause altogether, I suggest a rewrite similar to: "To familiarize yourself with the structure of a documentation component, see Directory Structure and Key Files."

File: modules/contribute/pages/create-branches.adoc (hunk 9 above)
Comment (polandll) [reply modules/contribute/pages/create-branches.adoc:15]: fixed

File: modules/contribute/pages/create-branches.adoc (hunk 10 above)
Comment (polandll) [reply modules/contribute/pages/create-branches.adoc:22]: fixed

File: modules/contribute/pages/install-git-and-editor.adoc [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+== Download and install Git
+
+To clone and work with Git repositories on your device, i.e., local repositories,
+install Git using your operating system's package manager.
+Alternatively, you can download it from the https://git-scm.com/downloads[Git project^].
```
[the remark this answers is not in the document]
Comment (polandll) [reply modules/contribute/pages/install-git-and-editor.adoc:15]: I don't think so, it's beyond the scope of this document. And honestly, if someone is planning to make contributions, they likely already have git installed.

File: modules/contribute/pages/set-up-repository.adoc (hunk 3 above)
Comment (polandll) [reply modules/contribute/pages/set-up-repository.adoc:36]: fixed

File: modules/contribute/pages/set-up-repository.adoc (hunk 4 above)
Comment (polandll) [reply modules/contribute/pages/set-up-repository.adoc:60]: added a return

File: modules/contribute/pages/repositories.adoc (hunk 6 above)
Comment (polandll) [reply modules/contribute/pages/repositories.adoc:19]: k8ssandra docs are not written in AsciiDoc, so these instructions don't apply to them.

File: modules/contribute/partials/component-terms.adoc (hunk 7 above)
Comment (polandll) [reply modules/contribute/partials/component-terms.adoc:19]: changed

File: modules/contribute/pages/create-branches.adoc [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
++
+image::project-panel.png[,300,align=left]
+
+. <<base-branch,Choose the right branch>> to base your changes on.
+Click the image:git-branch.svg[,16,role=icon] branch icon at the bottom of the editor.
+In the modal window, select the desired base branch from the dropdown menu.
++
+image::select-base-branch.png[,380,align=left]
+
+. Pull the latest changes from the upstream repository using the `fetch` at the bottom of the editor
```
[the remark this answers is not in the document]
Comment (polandll) [reply modules/contribute/pages/create-branches.adoc:37]: fixed

File: modules/contribute/pages/workflow-overview.adoc [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+same set of changes over several days before requesting sign-off.
+
+== Workflow Steps
+
+These steps, described under Setting Up Your Workspace, only need to be done
+once or the first time you plan to contribute to a repository:
+
+* [ ] Fork the repository into your GitHub account.
+* [ ] Create a clone of the repository on your device.
+* [ ] Connect your local repository with the upstream repository.
```
[the remark this answers is not in the document]
Comment (polandll) [reply modules/contribute/pages/workflow-overview.adoc:14]: added

[the remark this answers is not in the document]
Comment (polandll) [reply modules/contribute/pages/workflow-overview.adoc:14]: fixed

**Comment by polandll** [comment]
Moved the AsciiDoc How-to to its own repo, so I'm closing this PR to pull this material into Stargate. Will accomplish it with a multi-repo build.

**Pull request closed without merging**